Assigment3/EmployeeManagement.java

```diff
@@ -0,0 +1,307 @@
+import java.io.*;
+import java.util.*;
+
+/**
+ * EmployeeManagement class
+ * empList - stores the all data of employee from emp.csv
+ * empFile - BufferedReader instance used to read data from .csv
+ * latestIdAllocated - which initially set to last id allocated to employee
+ * location - represents file location initialized in constructor
+ * saveEmp - used to save data in file .csv
+ * changeStatus - it used to detect is there any changes made in data by add/delete/update operation
+ */
+class EmployeeManagement{
+    private HashMap<Integer,Employee> empList;
+    private BufferedReader empFile;
+    private int latestIdAllocated;
+    private File location;
+    private BufferedWriter saveEmp;
+    private boolean changeStatus; 
+
+    EmployeeManagement(){
+        changeStatus=false;
+        location=new File("./emp.csv");
+        getAllEmployeeData();
+        showAllEmployee();
+    }
+
+    /**Feaching all data from file and Storing in a empList DS */
+    public void getAllEmployeeData(){
+        this.empList=new HashMap<>();
+        this.latestIdAllocated=0;
+        try {
+            // connecting with file
+            FileReader fr=new FileReader(this.location);
+            // class instance
+            this.empFile=new BufferedReader(fr);
+
+            // started reading empdata
+            String empData=this.empFile.readLine(); 
+            while(empData!=null){
+                Employee tempEmp=getEmployeeByString(empData);
+                empList.put(tempEmp.getUid() , tempEmp);
+    
+                
+                this.latestIdAllocated=Math.max(this.latestIdAllocated,tempEmp.getUid());
+                // reading
+                empData=this.empFile.readLine();
+            }
+
+        } catch (Exception e) {
+             e.printStackTrace();
+        }
+        
+    }
+
+    /**Just converting String data into an Employee object */
+    private Employee getEmployeeByString(String tEmployee){
+        String empField[]=tEmployee.split(",");
+        int id=Integer.parseInt(empField[0]);
+        int age=Integer.parseInt(empField[3]);
+        Employee newEmp=new Employee(id,empField[1],empField[2],age,empField[4]);
+        return newEmp;
+    }
+    
+    /**Showing all Employee detail stored in CSV file */
+    public void showAllEmployee(){
+        System.out.println("All Employee Detail are showed here.....\n");
+        for(Employee temp:empList.values()){
+            System.out.println(temp);
```

Comment:
Please use self explanatory variable names, "temp" is not the right example of a variable name. It will help you in a long run. Please make it part of your programming activity.